pkg/controller/core/workload_controller.go

```diff
@@ -309,6 +313,9 @@ func (r *WorkloadReconciler) reconcileOnClusterQueueActiveState(ctx context.Cont
 		log.V(3).Info("Workload is evicted because the ClusterQueue is stopped", "clusterQueue", klog.KRef("", cqName))
 		workload.SetEvictedCondition(wl, kueue.WorkloadEvictedByClusterQueueStopped, "The ClusterQueue is stopped")
 		err := workload.ApplyAdmissionStatus(ctx, r.client, wl, true)
+		if err == nil {
+			metrics.ReportEvictedWorkloads(cqName, kueue.WorkloadEvictedByClusterQueueStopped)
```

Comment:
This one you could check [here](https://github.com/kubernetes-sigs/kueue/blob/36b340e396341c905444576bf797a221cb5f92ac/test/integration/scheduler/scheduler_test.go#L1778), by `util.ExpectEvictedWorkloadsTotalMetric(clusterQueue.Name, kueue.WorkloadEvictedByClusterQueueStopped, 1)`. 

However, it turns out that the metric counts as 2 at this point (I checked that), which suggests a bug, because we trigger the patch twice. 

The second time is called when the Evicted=true condition is already there, I think if the `Evicted` condition is already there, then we can skip sending the patch again (and thus bumping the metric). 

If this is the case, then the PR already revealed a performance improvement in this scenario. cc @alculquicondor

Reply:
Note how we avoid sending eviction request twice in case of preemption: https://github.com/kubernetes-sigs/kueue/blob/36b340e396341c905444576bf797a221cb5f92ac/pkg/scheduler/preemption/preemption.go#L156

Reply:
I have add a conditon check before do patch:
```go
		if apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted)  {
			log.V(3).Info("Workload is already evicted due to clusterQueue stopped.")
			return true, nil
		}
```